main/src/main/scala/sbt/Defaults.scala

```diff
@@ -474,16 +474,16 @@ object Defaults extends BuildCommon {
       Try(Paths.get(path)).map(_ => true).getOrElse(false)
     }
 
-    val newPath: Option[String] = pos
+    val absSource: Option[File] = pos
       .sourcePath()
       .asScala
       .filter(isValid)
       .map { path =>
-        fc.toPath(VirtualFileRef.of(path)).toAbsolutePath.toString
+        fc.toPath(VirtualFileRef.of(path)).toAbsolutePath.toFile
```

Comment:
I am not sure if this is the right move. Part of my motivation for introducing VirtualFile was to actually use it for in-memory files. They do not have to convert to `java.io.File`.

Reply:
I think this gets to what we can work with - https://github.com/sbt/sbt/pull/6367